quotes.py

```diff
@@ -0,0 +1,258 @@
+                    fwd_status = 'channel_private'
+            else:
+                fwd_status = 'open'
+
+        if fwd_status == 'open':
+            user_id = reply.fwd_from.from_id
+        elif fwd_status == 'no_fwd':
+            user_id = reply.from_id
+        elif fwd_status == 'channel_open':
+            resolved = await self.client(GetFullChannelRequest(reply.fwd_from.saved_from_peer.channel_id))
+            chat = resolved.chats[0]
+            user_id = chat.id
+        elif fwd_status == 'channel_open_special':
+            resolved = await self.client(GetFullChannelRequest(reply.fwd_from.channel_id))
+            chat = resolved.chats[0]
+            user_id = chat.id
+            fwd_status = 'channel_open'  # now it's just regular channel, only saved_from_peer was the difference.
+
+        if fwd_status not in ['hidden', 'channel_private']:
+            if not await self.client.get_profile_photos(user_id):
+                no_pfp = True
+            else:
+                no_pfp = False
+        else:
+            no_pfp = True
+
+        if fwd_status not in ['hidden', 'channel_private']:
+            pfp_bytes = BytesIO()
+            await self.client.download_profile_photo(user_id, file=pfp_bytes)
+            pfp_bytes.seek(0)
+            if no_pfp is False:
+                pfp = 'data:image/png;base64, ' + base64.b64encode(pfp_bytes.read()).decode()
+            else:
+                pfp = ''
+            pfp_bytes.close()
+
+        if fwd_status not in ['hidden', 'channel_private']:
+            colors = ["#fb6169", "#85de85", "#f3bc5c", "#65bdf3", "#b48bf2", "#ff5694", "#62d4e3", "#faa357"]
+            num1 = user_id % 7
+            num2 = [0, 7, 4, 1, 6, 3, 5]
+            colour = colors[num2[num1]]
+        else:
+            colour = '#b48bf2'
+
+        if fwd_status == 'no_fwd' and isinstance(message.chat, Channel):
+            participant = await self.client(GetParticipantRequest(message.chat_id, user_id))
+            user = participant.users[0]
+        elif fwd_status not in ['hidden', 'channel_open', 'channel_private']:
+            user = (await self.client(GetFullUserRequest(user_id))).user
+
+        if fwd_status == 'channel_open':
+            username = chat.title
+        elif fwd_status == 'hidden':
+            username = reply.fwd_from.from_name
+        elif fwd_status == 'channel_private':
+            username = 'Hidden.'
+        else:
+            username = user.first_name
+            if user.last_name:
+                username += f' {user.last_name}'
+
+        if fwd_status.startswith('channel'):
+            admintitle = 'channel'
+        elif fwd_status == 'hidden':
+            admintitle = ''
+        else:
+            try:
+                admintitle = participant.participant.rank
+                if not admintitle:
+                    admintitle = 'admin'
+            except (AttributeError, UnboundLocalError):
```

Comment:
Woah woah hold your horses. Wth is going on here? You should use getattr(participant, "participant", None) if you are worried there's no participant